tests/queries/0_stateless/03164_optimize_row_order_during_insert.sql

```diff
@@ -0,0 +1,10 @@
+-- Checks that no bad things happen when the table optimizes the row order to improve compressability during just simple insert.
+
+DROP TABLE IF EXISTS tab;
+
+CREATE TABLE tab (name String, event Int8) ENGINE = MergeTree ORDER BY name SETTINGS allow_experimental_optimized_row_order = true;
+INSERT INTO tab VALUES ('Igor', 3), ('Egor', 1), ('Egor', 2), ('Igor', 2), ('Igor', 1);
+
+SELECT * FROM tab SETTINGS max_threads=1;
+
+DROP TABLE tab;
```

Comment:
Could we please move all tests into a single test / expected results file 03164_optimize_row_order_during insert.sql?

To separate the individual expected results in the reference file visually, you may use this trick:

```sql
SELECT 'Simple test';
[...]

SELECT 'Cardinalities test';
[...]

etc.
```